pkg/registry/core/shoot/strategy_test.go

```diff
@@ -33,7 +34,53 @@ import (
 )
 
 var _ = Describe("Strategy", func() {
-	Context("PrepareForUpdate", func() {
+	Describe("#PrepareForCreate", func() {
+		Context("max token expiration", func() {
+			newShoot := func(duration time.Duration) *core.Shoot {
+				return &core.Shoot{
+					Spec: core.ShootSpec{
+						Kubernetes: core.Kubernetes{
+							KubeAPIServer: &core.KubeAPIServerConfig{
+								ServiceAccountConfig: &core.ServiceAccountConfig{
+									MaxTokenExpiration: &metav1.Duration{Duration: duration},
+								},
+							},
+						},
+					},
+				}
+			}
+
+			DescribeTable("ShootMaxTokenExpirationOverwrite feature gate enabled",
+				func(maxTokenExpiration, expectedDuration time.Duration) {
+					defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.ShootMaxTokenExpirationOverwrite, true)()
+
+					shoot := newShoot(maxTokenExpiration)
+					shootregistry.Strategy.PrepareForCreate(context.TODO(), shoot)
+					Expect(shoot.Spec.Kubernetes.KubeAPIServer.ServiceAccountConfig.MaxTokenExpiration.Duration).To(Equal(expectedDuration))
+				},
+
+				Entry("too low value", time.Hour, 720*time.Hour),
+				Entry("too high value", 3000*time.Hour, 2160*time.Hour),
+				Entry("value within boundaries", 1000*time.Hour, 1000*time.Hour),
+			)
+
+			DescribeTable("ShootMaxTokenExpirationOverwrite feature gate disabled",
+				func(maxTokenExpiration, expectedDuration time.Duration) {
+					defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.ShootMaxTokenExpirationOverwrite, false)()
+
+					shoot := newShoot(maxTokenExpiration)
+					shootregistry.Strategy.PrepareForCreate(context.TODO(), shoot)
+					Expect(shoot.Spec.Kubernetes.KubeAPIServer.ServiceAccountConfig.MaxTokenExpiration.Duration).To(Equal(expectedDuration))
+				},
+
+				Entry("too low value", time.Hour, time.Hour),
+				Entry("too high value", 3000*time.Hour, 3000*time.Hour),
+				Entry("value within boundaries", 1000*time.Hour, 1000*time.Hour),
+			)
```

Comment:
Sorry for the misunderstanding, I should have add more details. I was thinking more about combining everything in one `DescribeTable` like below. Also, the same for the tests starting from L477.
```suggestion
			DescribeTable("ShootMaxTokenExpirationOverwrite",
				func(maxTokenExpiration, expectedDuration time.Duration, featureGateEnabled bool) {
					defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.ShootMaxTokenExpirationOverwrite, featureGateEnabled)()

					shoot := newShoot(maxTokenExpiration)
					shootregistry.Strategy.PrepareForCreate(context.TODO(), shoot)
					Expect(shoot.Spec.Kubernetes.KubeAPIServer.ServiceAccountConfig.MaxTokenExpiration.Duration).To(Equal(expectedDuration))
				},

				Entry("too low value with feature gate enabled", time.Hour, 720*time.Hour, true),
				Entry("too high value with feature gate enabled", 3000*time.Hour, 2160*time.Hour, true),
				Entry("value within boundaries with feature gate enabled", 1000*time.Hour, 1000*time.Hour, true),
				Entry("too low value with feature gate disabled", time.Hour, 720*time.Hour, false),
				Entry("too high value with feature gate disabled", 3000*time.Hour, 2160*time.Hour, false),
				Entry("value within boundaries with feature gate disabled", 1000*time.Hour, 1000*time.Hour, false),
			)
```